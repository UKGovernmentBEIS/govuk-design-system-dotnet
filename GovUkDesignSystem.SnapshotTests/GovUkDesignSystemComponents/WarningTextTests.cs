using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class WarningTextTests : SnapshotTestBase
    {
        private WarningTextViewModel DefaultWarningTextViewModel()
        {
            return new WarningTextViewModel
            {
                IconFallbackText = "Warning",
                Text = "This is a warning!",
                Classes = "test-class",
                Attributes = new Dictionary<string, string>
                {
                    {"key", "value"}
                }
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("WarningText", DefaultWarningTextViewModel());
        }
        
        [Fact]
        public async Task Render_Html()
        {
            // Arrange
            var viewModel = DefaultWarningTextViewModel();
            viewModel.Text = null;
            viewModel.Html = o => "html";
            
            // Act & Assert
            await VerifyPartial("WarningText", viewModel);
        }
    }
}