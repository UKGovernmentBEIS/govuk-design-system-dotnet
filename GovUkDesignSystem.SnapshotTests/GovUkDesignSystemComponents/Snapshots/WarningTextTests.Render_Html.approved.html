<div class="govuk-warning-text test-class" key="value">
  <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
  <strong class="govuk-warning-text__text">
    <span class="govuk-warning-text__assistive">Warning</span>

html    </strong>
</div>